orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,307 @@
+		SkipTests.getSkipList("normal_event_skip_list.txt");
+		
+		// For now an array of species names are used. I will likely change it so that the wrapper calls each organism individually during release.
+		List<String> speciesList = new ArrayList<String>(Arrays.asList("pfal", "spom", "scer", "ddis", "cele", "sscr", "btau", "cfam", "mmus", "rnor", "ggal", "tgut", "xtro", "drer", "dmel", "atha", "osat"));
+
+		JSONParser parser = new JSONParser();
+		Object obj = parser.parse(new FileReader(pathToSpeciesConfig));
+		JSONObject jsonObject = (JSONObject) obj;
+		for (String species : speciesList)
+		{
+			// Proper logging will be implemented before Release
+			JSONObject speciesObject = (JSONObject) jsonObject.get(species);
+			JSONArray speciesNames = (JSONArray) speciesObject.get("name");
+			String speciesName = (String) speciesNames.get(0);
+			System.out.println("Beginning orthoinference of " + speciesName + ".");
+			JSONObject refDb = (JSONObject) speciesObject.get("refdb");
+			String refDbUrl = (String) refDb.get("url");
+			String refDbProteinUrl = (String) refDb.get("access");
+			String refDbGeneUrl = (String) refDb.get("ensg_access");
+
+			JSONObject altRefDb = (JSONObject) speciesObject.get("alt_refdb");
+
+			// Creates two files that a) list reactions that are eligible for inference and b) those that are successfully inferred
+			String eligibleFilename = "eligible_" + species	+ "_75.txt";
+			String inferredFilename = "inferred_" + species + "_75.txt";
+			PrintWriter eligibleFile = new PrintWriter(eligibleFilename);
+			PrintWriter inferredFile = new PrintWriter(inferredFilename);
+			eligibleFile.close();
+			inferredFile.close();
+			InferReaction.setEligibleFilename(eligibleFilename);
+			InferReaction.setInferredFilename(inferredFilename);
+
+			// Set static variables (DB/Species Instances, mapping files) that will be repeatedly used
+			InferEvents.setInstanceEdits();
+			try {
+				HashMap<String,String[]> homologueMappings = InferEvents.readHomologueMappingFile(species, "hsap");
+				ProteinCount.setHomologueMappingFile(homologueMappings);
+				InferEWAS.setHomologueMappingFile(homologueMappings);
+				homologueMappings = new HashMap<String,String[]>();
+			} catch (FileNotFoundException e) {
+				System.out.println("Unable to locate " + speciesName +" mapping file: hsap_" + species + "_mapping.txt. Orthology prediction not possible.");
+				continue;
+			}
+			InferEWAS.readENSGMappingFile(species);
+			InferEWAS.createUniprotDbInst();
+			InferEWAS.createEnsemblProteinDbInst(speciesName, refDbUrl, refDbProteinUrl);
+			InferEWAS.createEnsemblGeneDBInst(speciesName, refDbUrl, refDbGeneUrl);
+			if (altRefDb != null)
+			{
+				JSONArray altRefDbNames = (JSONArray) altRefDb.get("dbname");
+				String altRefDbUrl = (String) altRefDb.get("url");
+				String altRefDbAccess = (String) altRefDb.get("access");
+				InferEWAS.createAlternateReferenceDBInst(speciesName, (String) altRefDbNames.get(0), altRefDbUrl, altRefDbAccess);
```

Comment:
This method might be better with a single parameter of altRefDb rather than decomposing it.